src/endpoint.jl

```diff
@@ -55,7 +92,16 @@ function Base.hash(x::Endpoint{T,D,B}, h::UInt) where {T,D,B}
     return h
 end
 
+function Base.hash(x::HalfOpenEndpoint{T,B}, h::UInt) where {T,B}
+    h = hash(:HalfOpenEndpoint, h)
+    h = hash(B, h)
+    return hash(x.endpoint, h)
+end
+
 Base.broadcastable(e::Endpoint) = Ref(e)
+Base.broadcastable(e::HalfOpenEndpoint) = Ref(e)
+
+offset_value(x::AbstractEndpoint) = isleft(x) ? !isclosed(x) : isclosed(x)
```

Comment:
Function name isn't descriptive. At a minimum needs a comment explaining the purpose